assembly/fs/async.ts

```diff
@@ -0,0 +1,26 @@
+import { Maybe, MaybeCallbackContext } from "../managed/maybe";
+import { Box } from "../message";
+import { writeFile } from "./sync";
+
+export class WriteWhileMaybeContext<T> {
+  constructor(
+    public path: string,
+    public contents: T,
+    public flags: string,
+    public encoding: string,
+  ) {}
+}
+
+export function writeWhileMaybe<T>(path: string, contents: T, flags: string = "w", encoding: string = "utf8"): Maybe<usize, string> {
+  let ctx = new WriteWhileMaybeContext<T>(path, contents, flags, encoding);
```

Comment:
Don't need to make this this mutable. Possibly one of the functions needed to modify ctx or it was just a mistake